CRM/Core/Block.php

```diff
@@ -349,14 +347,12 @@ private static function setTemplateShortcutValues() {
 
       $components = CRM_Core_Component::getEnabledComponents();
 
-      if (!empty($config->enableComponents)) {
+      if ($components) {
         // check if we can process credit card contribs
         $newCredit = CRM_Core_Config::isEnabledBackOfficeCreditCardPayments();
 
-        foreach ($components as $componentName => $obj) {
-          if (in_array($componentName, $config->enableComponents)) {
-            $obj->creatNewShortcut($shortCuts, $newCredit);
-          }
+        foreach ($components as $obj) {
+          $obj->creatNewShortcut($shortCuts, $newCredit);
```

Comment:
I had to read this code a couple times to make sense of it. Basically it was fetching a list of enabled components, then for each one, checking to see if it's enabled before taking an action. That's obviously redundant, so I removed the extra check.